### PR TITLE
Added support for gethostbyname, for passing host names in addition to I...

### PR DESCRIPTION
**Pull request by frederickproctor**
...P addresses.

### CODE REVIEW

**Comment by shaun-edwards** [comment]
Thanks for the contribution!

One small modification is requested.  For some history, this simple message library tries to be OS independent (at one time it ran on the Motoman controller, and in the future it would be nice to use it on Windows).

All system calls are wrapped in macros (for linux see [here](https://github.com/frederickproctor/industrial_core/blob/gethostbyname/simple_message/include/simple_message/socket/simple_socket.h#L54-L71), for motoplus see [here](https://github.com/frederickproctor/industrial_core/blob/gethostbyname/simple_message/include/simple_message/socket/simple_socket.h#L81-L112)) to achieve this OS independence.  

Could you please wrap the call to `gethostbyname` in a macro.  The linux version would perform the call the way you have it.  The motoplus version would simply return `NULL` since I don't believe that it is supported.

Other style comments in line.

File: simple_message/include/simple_message/socket/tcp_socket.h [hunk 1]
```diff
@@ -42,6 +42,7 @@
 
 #ifdef LINUXSOCKETS
 #include "sys/socket.h"
+#include "netdb.h"		/* gethostbyname */
```
Comment (shaun-edwards) [inline simple_message/include/simple_message/socket/tcp_socket.h]: No need to have the comment `/* gethostbyname */`or to at least match the current style.

File: simple_message/src/socket/tcp_client.cpp [hunk 2]
```diff
@@ -70,10 +72,16 @@ bool TcpClient::init(char *buff, int port_num)
       LOG_WARN("Failed to set no socket delay, sending data can be delayed by up to 250ms");
     }
 
+
     // Initialize address data structure
     memset(&this->sockaddr_, 0, sizeof(this->sockaddr_));
     this->sockaddr_.sin_family = AF_INET;
-    this->sockaddr_.sin_addr.s_addr = INET_ADDR(buff);
+    if (NULL != (ent = gethostbyname(buff))) {
```
Comment (shaun-edwards) [inline simple_message/src/socket/tcp_client.cpp]: Looks like the convention in this file is to put the `{` on the next line.

File: simple_message/src/socket/tcp_client.cpp [hunk 3]
```diff
@@ -70,10 +72,16 @@ bool TcpClient::init(char *buff, int port_num)
...
 
+
     // Initialize address data structure
     memset(&this->sockaddr_, 0, sizeof(this->sockaddr_));
     this->sockaddr_.sin_family = AF_INET;
-    this->sockaddr_.sin_addr.s_addr = INET_ADDR(buff);
+    if (NULL != (ent = gethostbyname(buff))) {
+      in_a = (struct in_addr *) ent->h_addr_list[0];
+      this->sockaddr_.sin_addr.s_addr = in_a->s_addr;
+    } else {
```
Comment (shaun-edwards) [inline simple_message/src/socket/tcp_client.cpp]: Same here for `{`

Comment (frederickproctor) [inline simple_message/src/socket/tcp_client.cpp]: Thanks for your coding comments, Shaun. I made the edits you noted, to wit: 
1. Macro'ed out gethostbyname in both the Linux and Motoman sections of simple_socket.h, with the Motoman resulting in NULL. Called that macro in tcp_client.c.
2. Took out the /\* gethostbyname*/ comment in the header, for consistency. 
3. Fixed the brace style in tcp_client.c, to be a new line for braces. I updated my emacs indenting accordingly. 
I think the first pull request is still pending, and you should see this new commit. 
--Fred

**Comment by shaun-edwards** [comment]
Thanks for the contribution @frederickproctor.  +1.

**Pull request merged**